File: 프로그래머스/lv2/42584. 주식가격/주식가격.java
class Solution {
    //가격이 떨어지지 않은 기간?
    public int[] solution(int[] prices) {

        int[] answer = new int[prices.length];
        for(int i=0; i<prices.length; i++) {
            for(int j=i+1; j< prices.length; j++) {
                answer[i]++;
                if(prices[i] > prices[j]) {
                    //떨어지면
                    break;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv2/42584. 주식가격/주식가격.py
from collections import deque

def solution(prices):
    result = [0] * len(prices)
    
    for i in range(0, len(prices)-1):
        for j in range(i+1, len(prices)):
            result[i] += 1
            if prices[i] > prices[j]:
                break;
    return result

        
def solution2(prices):
    answer = []
    q = deque(prices)
    
    while q:
        price = q.popleft()
        sec = 0
        
        for value in q:
            sec += 1
            if price > value:
                break;
        
        answer.append(sec)
    
    return answer